src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -34,162 +44,541 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+                    break;
+
+                case SymbolKind.Namespace:
+                    yield return Diagnostic.Create(NamespaceRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString());
+                    break;
+
+                case SymbolKind.NamedType:
+                    foreach (var location in symbol.Locations)
+                    {
+                        yield return Diagnostic.Create(TypeRule, location, misspelledWord, symbol.ToDisplayString());
+                    }
+
+                    break;
+
+                case SymbolKind.Method:
+                case SymbolKind.Property:
+                case SymbolKind.Event:
+                case SymbolKind.Field:
+                    yield return Diagnostic.Create(MemberRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString());
+                    break;
+
+                case SymbolKind.Parameter:
+                    yield return symbol.ContainingType.TypeKind == TypeKind.Delegate
+                        ? Diagnostic.Create(DelegateParameterRule, symbol.Locations.First(), symbol.ContainingType.ToDisplayString(), misspelledWord, symbol.Name)
+                        : Diagnostic.Create(MemberParameterRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), misspelledWord, symbol.Name);
+
+                    break;
+
+                case SymbolKind.TypeParameter:
+                    yield return symbol.ContainingSymbol.Kind == SymbolKind.Method
+                        ? Diagnostic.Create(MethodTypeParameterRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), misspelledWord, symbol.Name)
+                        : Diagnostic.Create(TypeTypeParameterRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), misspelledWord, symbol.Name);
+
+                    break;
+
+                case SymbolKind.Local:
+                    yield return Diagnostic.Create(VariableRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString());
+                    break;
+
+                default:
+                    throw new NotImplementedException($"Unknown SymbolKind: {symbol.Kind}");
+            }
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext compilationStartContext)
+        {
+            var projectDictionary = _mainDictionary.Clone();
+            var dictionaries = ReadDictionaries();
+            if (dictionaries.Any())
+            {
+                var aggregatedDictionary = dictionaries.Aggregate((x, y) => x.CombineWith(y));
+                projectDictionary = projectDictionary.CombineWith(aggregatedDictionary);
+            }
+
+            compilationStartContext.RegisterOperationAction(AnalyzeVariable, OperationKind.VariableDeclarator);
+            compilationStartContext.RegisterCompilationEndAction(AnalyzeAssembly);
```

Comment:
❕ This analyzer should not include a compilation end action. Assembly diagnostics need to be reported elsewhere.

Reply:
There is no other alternative - assembly level diagnostics have to go into compilation level actions. We already have many analyzers doing such. They will be build-only diagnostics, but that is fine.

Reply:
Kept it